consumer/consumer.go

```diff
@@ -81,6 +81,12 @@ func NewWithSaramaConfig(
 	if saramaConfig == nil {
 		saramaConfig = sarama.NewConfig()
 		saramaConfig.Version = sarama.V0_10_2_0
+
+		if brokerCfg.Timeout > 0 {
```

Comment:
It's applied only to the default config(if there was no saramaConfig passed. Is it expected behaviour?

Reply:
I think so, because if you are passing a `sarama.Config` you can define the timeouts directly without using our configuration at all.

Anyway, we are always using the default configuration at the moment.